From 1c95d2afda3f7ad459de0a59e5e0bb09c9fb9148 Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Sun, 5 May 2019 02:30:11 +0200
Subject: [PATCH 59/98] ARM: dts: meson8: add the HDMI controller - WiP

WiP

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>

[Rebased for kernel 6.12.96: this patch originally anchored the new
hdmi_hpd_pins/hdmi_i2c_pins nodes right before i2c_b_pins, which does
not exist on this kernel version (same situation as patch 036/059's
pwm_c_dv9_pins/pwm_d_pins) — inserted them right after gpio-ranges
instead, immediately before pwm_c_dv9_pins, and folded in the
bias-disable removal from those two pwm pin groups as part of the same
hunk.]
---
 arch/arm/boot/dts/amlogic/meson8.dtsi | 66 ++++++++++++++++++++++++++-
 1 file changed, 64 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/meson8.dtsi b/arch/arm/boot/dts/amlogic/meson8.dtsi
index 9e4b6554298a3e..27262c685fcf02 100644
--- a/arch/arm/boot/dts/amlogic/meson8.dtsi
+++ b/arch/arm/boot/dts/amlogic/meson8.dtsi
@@ -315,6 +315,39 @@
 			#cooling-cells = <2>; /* min followed by max */
 		};
 
+		hdmi_tx: hdmi-tx@42000 {
+			compatible = "amlogic,meson8-hdmi-tx";
+			reg = <0x42000 0xc>;
+			interrupts = <GIC_SPI 57 IRQ_TYPE_EDGE_RISING>;
+			phys = <&hdmi_tx_phy>;
+			phy-names = "hdmi";
+			clocks = <&clkc CLKID_HDMI_PCLK>,
+				 <&clkc CLKID_HDMI_SYS>;
+			clock-names = "pclk", "sys";
+
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			#sound-dai-cells = <1>;
+			sound-name-prefix = "HDMITX";
+
+			status = "disabled";
+
+			/* VPU VENC Input */
+			hdmi_tx_venc_port: port@0 {
+				reg = <0>;
+
+				hdmi_tx_in: endpoint {
+					remote-endpoint = <&hdmi_tx_out>;
+				};
+			};
+
+			/* TMDS Output */
+			hdmi_tx_tmds_port: port@1 {
+				reg = <1>;
+			};
+		};
+
 		vpu: vpu@100000 {
 			compatible = "amlogic,meson8-vpu";
 
@@ -378,6 +411,15 @@
 			cvbs_vdac_port: port@0 {
 				reg = <0>;
 			};
+
+			/* HDMI-TX output port */
+			hdmi_tx_port: port@1 {
+				reg = <1>;
+
+				hdmi_tx_out: endpoint {
+					remote-endpoint = <&hdmi_tx_in>;
+				};
+			};
 		};
 	};
 }; /* end of / */
@@ -544,11 +544,26 @@
 			gpio-ranges = <&pinctrl_cbus 0 0 120>;
 		};
 
+		hdmi_hpd_pins: hdmi-hpd {
+			mux {
+				groups = "hdmi_hpd";
+				function = "hdmi";
+				bias-disable;
+			};
+		};
+
+		hdmi_i2c_pins: hdmi-i2c {
+			mux {
+				groups = "hdmi_sda", "hdmi_scl";
+				function = "hdmi";
+				bias-disable;
+			};
+		};
+
 		pwm_c_dv9_pins: pwm-c-dv9 {
 			mux {
 				groups = "pwm_c_dv9";
 				function = "pwm_c";
-				bias-disable;
 			};
 		};
 
@@ -556,7 +571,6 @@
 			mux {
 				groups = "pwm_d";
 				function = "pwm_d";
-				bias-disable;
 			};
 		};

@@ -760,6 +815,13 @@
 		assigned-clocks = <&clkc CLKID_VPU>;
 		assigned-clock-rates = <364285714>;
 	};
+
+	hdmi_tx_phy: hdmi-phy@3a0 {
+		compatible = "amlogic,meson8-hdmi-tx-phy";
+		clocks = <&clkc CLKID_HDMI_PLL_HDMI_OUT>;
+		reg = <0x3a0 0xc>;
+		#phy-cells = <0>;
+	};
 };
 
 &hwrng {

